apps/web/src/server/api/routers/product.ts

```diff
@@ -36,4 +36,27 @@ export const productRouter = createTRPCRouter({
 				nextCursor,
 			};
 		}),
+
+	searchProductCatalog: publicProcedure
+		.input(
+			z.object({
+				name: z.string(),
+			}),
+		)
+		.query(({ input }) => {
+			const { name } = input;
+
+			// matching product list
+			const products = [];
+
+			for (const product of mockedProducts) {
```

Comment:
Please consider using array functions instead of for loops https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array